data-prepper-plugins/http-source-common/src/main/java/org/opensearch/dataprepper/http/HttpServerConfig.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.http;
+
+import org.opensearch.dataprepper.model.types.ByteCount;
+import org.opensearch.dataprepper.model.configuration.PluginModel;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.micrometer.core.instrument.util.StringUtils;
+import jakarta.validation.constraints.AssertTrue;
+import jakarta.validation.constraints.Min;
+
+/**
+ * HttpServerConfig class holds the common http configurations defined in http source.
+*/
+public class HttpServerConfig {
```

Comment:
Let's make this an interface. Then you can create a `BaseHttpServerConfig` that implements the interface and has the same implementation that you have here.

We may find that we want a different set of default values or even different configurations between sources. So decoupling the usage of this interface from the concrete configuration will allow these to vary as needed.